src/Sets/HParallelotope.jl

```diff
@@ -0,0 +1,311 @@
+export HParallelotope,
+       directions,
+       offset,
+       dim,
+       base_vertex,
+       extremal_vertices,
+       center,
+       genmat,
+       generators,
+       constraints_list
+
+"""
+    HParallelotope{N<:Real, VN<:AbstractVector{N}, MN<:AbstractMatrix{N}} <: AbstractZonotope{N}
+
+Type that represents a parallelotope in constraint form.
+
+### Fields
+
+- `directions` -- square matrix such that each row is the direction of two parallel constraints
+- `offset`     -- vector such that each element is the offset of the corresponding constraint
+
+### Notes
+
+Parallelotopes are centrally symmetric convex polytopes in ``\\mathbb{R}^n`` having
+``2n`` pairwise parallel constraints. Every parallelotope is a zonotope.
+As such, parallelotopes can be represented either in constraint form or in
+generator form. The `HParallelotope` type represents parallelotopes in constraint form.
+
+The constraint form of a parallelotope is described next. Let
+``D ∈ \\mathbb{R}^{n × n}`` be a matrix and let ``c ∈ \\mathbb{R}^{2n}`` be
+a vector. The parallelotope ``P ⊂ \\mathbb{R}^n`` generated by the directions
+matrix ``D`` and the offset vector ``c`` is given by the set of points
+``x ∈ \\mathbb{R}^{n}`` such that:
+
+```math
+    D_i ⋅ x ≤ c_{i},\\text{  and  } -D_i ⋅ x ≤ c_{n+i}
+```
+for ``i = 1, …, n``. Here ``D_i`` represents the ``i``-th row of ``D`` and ``c_i`` the ``i``-th
+component of ``c``.
+
+For details on the notions given in these notes as well as applications of parallelotopes in
+reachability analysis we refer to [1] and [2]. For conversions between set representations
+we refer to [3].
+
+### References
+
+[1] Tommaso Dreossi, Thao Dang, and Carla Piazza. *Reachability computation for polynomial dynamical systems.*
+    Formal Methods in System Design 50.1 (2017): 1-38.
+
+[2] Tommaso Dreossi, Thao Dang, and Carla Piazza. *Parallelotope bundles for polynomial reachability.*
+    Proceedings of the 19th International Conference on Hybrid Systems: Computation and Control. ACM, 2016.
+
+[3] Matthias Althoff, Olaf Stursberg, and Martin Buss. *Computing reachable sets of hybrid systems using
+    a combination of zonotopes and polytopes.* Nonlinear analysis: hybrid systems 4.2 (2010): 233-249.
+"""
+struct HParallelotope{N<:Real, VN<:AbstractVector{N}, MN<:AbstractMatrix{N}} <: AbstractZonotope{N}
```

Comment:
i would suggest to remove the type annotation `N<:Real` here. (we do have this kind of restriction the LazySets but this new `HParallelotope` type should not force using `Real`). @schillic  do you agree?

Reply:
Yes, sure.